api/v1/common_types.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2022 Tigera, Inc. All rights reserved.
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	v1 "k8s.io/api/core/v1"
+)
+
+// Container is an arbitrary Kubernetes container.
+// Note: this should not be used directly as a field type within a component resource type.
+// Component resources should define their own container types with validation.
+type Container struct {
```

Comment:
Mentioned in another thread - what do you think about getting rid of this intermediate struct and instead using a v1.Container? 

I think it might make things a bit simpler. Given this struct isn't actually part of our API, I think it's OK and will remove the need to define a new structure.

Reply:
a96a441d9cc159e1c9f145ec053523222fdb42a7